google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/readrows/ReadRowsRetryTest.java

```diff
@@ -124,6 +134,98 @@ public void immediateRetryTest() {
     Truth.assertThat(actualResults).containsExactly("k1", "r1", "r2").inOrder();
   }
 
+  public ApiException largeRowExceptionWithTrailers(String rowKey) {
+    ErrorInfo errorInfo =
+        ErrorInfo.newBuilder()
+            .setReason("LargeRowReadError")
+            .setDomain("bigtable.googleapis.com")
+            .putMetadata("rowKey", rowKey)
+            .build();
+
+    Any packedErrorInfo = Any.pack(errorInfo);
+    // ErrorDetails errorDetails = ErrorDetails.builder()
+    //     .setRawErrorMessages(Collections.singletonList(packedErrorInfo))
+    //     .build();
+
+    ErrorDetails errorDetails =
+        ErrorDetails.builder().setRawErrorMessages(ImmutableList.of(packedErrorInfo)).build();
+
+    Metadata trailers = new Metadata();
+    byte[] status =
+        com.google.rpc.Status.newBuilder().addDetails(Any.pack(errorInfo)).build().toByteArray();
+    trailers.put(ERROR_DETAILS_KEY, status);
+    return (ApiException)
```

Comment:
You don't need to cast, UnavailableException is a ApiException https://github.com/googleapis/sdk-platform-java/blob/main/gax-java/gax/src/main/java/com/google/api/gax/rpc/UnavailableException.java#L36